app/src/main/java/is/xyz/mpv/MPVActivity.java

```diff
@@ -268,22 +290,26 @@ public void updateDecoderButton() {
     }
 }
 
-class HideControlsRunnable implements Runnable {
-    private View controls;
-    private View decorView;
+class FadeOutControlsRunnable implements Runnable {
+    private final View controls;
 
-    public HideControlsRunnable(View controls_, View decorView_) {
+    public FadeOutControlsRunnable(View controls_) {
         super();
         controls = controls_;
-        decorView = decorView_;
     }
 
-    @Override public void run() {
+    @Override
+    public void run() {
         // use GONE here instead of INVISIBLE (which makes more sense) because of Android bug with surface views
         // see http://stackoverflow.com/a/12655713/2606891
-        controls.setVisibility(View.GONE);
-
-        int flags = View.SYSTEM_UI_FLAG_HIDE_NAVIGATION | View.SYSTEM_UI_FLAG_FULLSCREEN;
-        decorView.setSystemUiVisibility(flags);
```

Comment:
Why is this not called here anymore? this results in back/home/square buttons not going away when controls disappear